services/catalog/src/simcore_service_catalog/models/domain/service.py

```diff
@@ -0,0 +1,277 @@
+            # remove the title of properties
+            for prop in schema.get("properties", {}).values():
+                prop.pop("title", None)
+
+
+class ServiceInput(ServiceProperty):
+    widget: Optional[Widget] = Field(
+        None,
+        description="custom widget to use instead of the default one determined from the data-type",
+    )
+
+
+class Service(BaseModel):
+    key: constr(regex=KEY_RE) = Field(
+        ...,
+        title="",
+        description="distinctive name for the node based on the docker registry path",
+        # regex=KEY_RE,
+        example=[
+            "simcore/services/comp/itis/sleeper",
+            "simcore/services/dynamic/3dviewer",
+        ],
+    )
+    version: constr(regex=VERSION_RE) = Field(
+        ...,
+        description="service version number",
+        # regex=VERSION_RE,
+        example=["1.0.0", "0.0.1"],
+    )
+    integration_version: Optional[constr(regex=VERSION_RE)] = Field(
+        None,
+        description="service oSparc integration version number",
+        # regex=VERSION_RE,
+        example="1.0.0",
+    )
+    service_type: ServiceType = Field(
+        ..., alias="type", description="service type", example="computational",
+    )
+    name: str = Field(
+        ...,
+        description="short, human readable name for the node",
+        example="Fast Counter",
+    )
+    thumbnail: Optional[HttpUrl] = Field(
+        None,
+        description="url to the thumbnail",
+        example="https://user-images.githubusercontent.com/32800795/61083844-ff48fb00-a42c-11e9-8e63-fa2d709c8baf.png",
+    )
+    badges: Optional[List[Badge]] = Field(None)
+    description: str = Field(
+        ...,
+        description="human readable description of the purpose of the node",
+        example=[
+            "Our best node type",
+            "The mother of all nodes, makes your numbers shine!",
+        ],
+    )
+    authors: List[Author] = Field(..., min_items=1)
+    contact: EmailStr = Field(
+        ...,
+        description="email to correspond to the authors about the node",
+        example=["lab@net.flix"],
+    )
+    inputs: Optional[Dict[PropertyName, ServiceInput]] = Field(
+        ..., description="all the input configurable for this service"
+    )
+    outputs: Optional[Dict[PropertyName, ServiceProperty]] = Field(
+        ..., description="all the output configurable for this service"
+    )
+
+    class Config:
+        description = "Description of a simcore node 'class' with input and output"
+        title = "simcore node"
+        extra = Extra.forbid
+
+        @staticmethod
+        def schema_extra(schema: Dict[str, Any], model: Type) -> None:
+            # remove the title of properties
+            for prop in schema.get("properties", {}).values():
+                prop.pop("title", None)
+
+
+if __name__ == "__main__":
+
+    with open(current_file.with_suffix(".json"), "wt") as fh:
+        print(Service.schema_json(indent=2), file=fh)
```

Comment:
I wonder if there is a way to compare json schemas

Reply:
there is